include/RAJA/policy/hip/MemUtils_HIP.hpp

```diff
@@ -113,9 +113,29 @@ struct DeviceZeroedAllocator {
   }
 };
 
+//! Allocator for device pinned memory for use in basic_mempool
+struct DevicePinnedAllocator {
+
+  // returns a valid pointer on success, nullptr on failure
+  void* malloc(size_t nbytes)
+  {
+    void* ptr;
+    hipErrchk(hipMalloc(&ptr, nbytes));
+    return ptr;
+  }
+
+  // returns true on success, false on failure
```

Comment:
```suggestion
  // returns true on success, throws a run time error exception on failure
```

https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/hip/raja_hiperrchk.hpp#L45